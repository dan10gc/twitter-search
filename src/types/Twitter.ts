export interface HashtagModel {
    text: string;
    indices: Array<number>;
}

export interface MetaDataModel {
    completed_in: number;
    count: number;
    max_id: number;
    max_id_str: string;
    // is not returned when user reaches end of list
    next_results?: string;
    query: string
    since_id: number
    since_id_str: string
}