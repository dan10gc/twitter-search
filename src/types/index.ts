export * from './Status'
export * from './Twitter';