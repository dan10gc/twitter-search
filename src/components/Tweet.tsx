import React from 'react';
// @ts-ignore
import extractUrls from 'extract-urls';
import Hashtag from './Hashtag';
import { HashtagModel } from '../types';

interface Props {
	tweet: any;
	index: number;
}

const Tweet = ({ tweet, index }: Props) => {
	const tweetUrl = extractUrls(tweet.text);
	const hashtags: Array<string> = tweet.entities.hashtags.map((hashtag: HashtagModel) => hashtag.text);
	const text = tweet.text.replace(/(?:https?|ftp):\/\/[\n\S]+/g, '');

	return (
		<div
			className={`box mb-0 px-mobile is-shadowless is-radiusless ${
				index % 2 !== 0 && `has-background-white-bis`
			}`}
		>
			<article className="media">
				<div className="media-left">
					<figure className="image is-48x48">
						<img className="is-rounded" src={tweet.user.profile_image_url} alt="user profile" />
					</figure>
				</div>
				<div className="media-content">
					<div className="content">
						<h5 className="title is-5 mb-2">
							<strong>@{tweet.user.screen_name}</strong>
						</h5>
						<p className="mb-1 has-text-weight-medium tweet-text">{text}</p>
						{Array.isArray(tweetUrl) && (
							<a href={tweetUrl[0]} target="_blank" rel="noreferrer">
								{tweetUrl[0]}
							</a>
						)}
					</div>
					<div className="is-flex-direction-row flex-wrap is-justify-content-start">
						{hashtags.map((hashtag: string) => (
							<Hashtag {...{ hashtag }} />
						))}
					</div>
				</div>
			</article>
		</div>
	);
};

export default Tweet;
