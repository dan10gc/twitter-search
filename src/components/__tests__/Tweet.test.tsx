import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import Tweet from '../Tweet';
import { Status } from '../../types';

const onLoadMoreMock = jest.fn();

const emptyTweetsArr: Array<any> = [];
// Ideally we move this to __mocks__
const mockTweet = {
	entities: { hashtags: [{ text: 'Messi', indices: [75, 101] }] },
	text: '💪 #Messi is back! 🐐 https://t.co/QyEDYPTNp7',
	user: {
		screen_name: 'FCBarcelona',
		profile_image_url: 'http://pbs.twimg.com/profile_images/1333096463916797954/7bzarkH2_normal.jpg',
	},
};

describe('Tweet Component', () => {
	test('renders tweet with white-bis background-color', () => {
		const { container, getByText } = render(<Tweet tweet={mockTweet} index={3} />);
		expect(getByText('#Messi')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
