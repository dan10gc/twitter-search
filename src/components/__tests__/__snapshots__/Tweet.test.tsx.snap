// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet Component renders tweet with white-bis background-color 1`] = `
<div>
  <div
    class="box mb-0 px-mobile is-shadowless is-radiusless has-background-white-bis"
  >
    <article
      class="media"
    >
      <div
        class="media-left"
      >
        <figure
          class="image is-48x48"
        >
          <img
            alt="user profile"
            class="is-rounded"
            src="http://pbs.twimg.com/profile_images/1333096463916797954/7bzarkH2_normal.jpg"
          />
        </figure>
      </div>
      <div
        class="media-content"
      >
        <div
          class="content"
        >
          <h5
            class="title is-5 mb-2"
          >
            <strong>
              @
              FCBarcelona
            </strong>
          </h5>
          <p
            class="mb-1 has-text-weight-medium tweet-text"
          >
            üí™ #Messi is back! üêê 
          </p>
          <a
            href="https://t.co/QyEDYPTNp7"
            rel="noreferrer"
            target="_blank"
          >
            https://t.co/QyEDYPTNp7
          </a>
        </div>
        <div
          class="is-flex-direction-row flex-wrap is-justify-content-start"
        >
          <div
            class="badge is-clickable"
          >
            <span>
               #
              Messi
            </span>
          </div>
        </div>
      </div>
    </article>
  </div>
</div>
`;
